packages/cli/src/commands/hydrogen/build.ts

```diff
@@ -197,4 +258,16 @@ export async function runBuild({
       );
     }
   }
+
+  return {
+    async close() {
+      codegenProcess?.removeAllListeners('close');
+      codegenProcess?.kill('SIGINT');
+
+      const promises: Array<Promise<void>> = [];
+      if ('close' in clientBuild) promises.push(clientBuild.close());
+      if ('close' in serverBuild) promises.push(serverBuild.close());
```

Comment:
The type of `clientBuild` and `serverBuild` is different in watch mode. Using `in` here correctly discriminates the type.